### ISSUE TITLE
Staging Review finding: Add option to remove other service names

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#82849]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Pension benefits
**Product:** 527EZ
**Feature:** Update multi-response (single page list loop) pattern

## Findings details
**VA.gov Experience Standard - issue:** [User can't determine next steps in a flow.](https://design.va.gov/about/experience-standards/#user-can-determine-next-steps-in-a-flow)
**VA.gov Experience Standard - category:** [Findability](https://design.va.gov/about/experience-standards/#category-findability)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @allison0034 (Design), @erinrwhite (IA)

### Description

On the /military/general/add page, when a user adds and saves a name there is no option to delete/remove the name. This could be a problem if they added a name by mistake. After looking thru the other multiple responses pattern, this is also missing on the VA medical, Current employment, Dependent children and Gross monthly income patterns.

<details><summary>Link, screenshot or steps to recreate</summary>

![Screenshot of card](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/74306-Allison-01.png?raw=true)

</details>

## Recommended action

Add a remove button to the card when in edit mode. You can review the guidance in the [Design System](https://design.va.gov/patterns/ask-users-for/multiple-responses#review-for-edit-and-remove). Be sure to include a confirmation modal for removing.

## References

- **Accessibility Defect Severity:** N/A
- **WCAG Success Criteria:** N/A
- **Modality:** N/A
- **Design System Component:** N/A
- **Design System Pattern or Template:** [Multiple responses](https://design.va.gov/patterns/ask-users-for/multiple-responses)
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** User flow

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by danbrady** [comment]
Figma File: https://www.figma.com/design/9JKK5Eo43uJWEr66JPiebc/Pension-Claim-Form-21P-527EZ?node-id=5366-16247&t=BkxIMm2hsm8YyJe1-1

Mock form example that shows validation adjusting previous answer based on removal of all added items: https://staging.va.gov/mock-form-patterns/employers

**Note:** These same updates apply to VA Medical Centers and Current Employment as well. The changes are the same, and are shown in the same Figma link above. (The staging review ticket also mentions applying these updates to Dependent Children and Gross Monthly Income as well. However, those steps are being updated from other issues so these changes do not apply to them.)

cc: @aplatt-coforma @matt4su @TaiWilkin

**Comment by aplatt-coforma** [comment]
Thanks @danbrady for providing the update designs for implementation.
Moving this to Product Backlog column for eng to work next sprint.

**Comment by ToddWebDev** [comment]
I have a question about this proposal:
This page only shows up when the Veteran has said that they 'have served under another name' on the previous page, so what exactly is supposed to happen after they remove this required service name? The single page array pattern does not have an option to remove the first item because it is a required item. There is a 'Remove' button on subsequent items since they are not required. If the Veteran chooses not to add another name they can choose 'Back' and edit the previous page. Perhaps the 'Remove' button in this scenario can serve as a 'Back' button?
https://www.figma.com/design/9JKK5Eo43uJWEr66JPiebc/Pension-Claim-Form-21P-527EZ?node-id=5366-16304&t=ZcuEzwC23qqDUC26-0

**Comment by danbrady** [comment]
Good question, Todd. It is a little odd.

My thinking was to keep any button actions consistent across cards. If there is only one card, clicking "remove" would remove that card, leaving the buttons "add another..." and continue/back[1]. (A bit of a awkward display, admittedly, but does what the user expects.)

If the user has two cards, they should be able to delete either or both. So if they removed card #1 first, they remain on that page now with a removal confirmation banner and card #2. If they then removed card #2 (the only one left), they should remain on that page. I'd be concerned that navigating to a different page upon clicking "remove" may be slightly disorienting.

So it's kind of a question of what is least awkward: a mostly empty "add" page, or navigating a user when they remove the last card? (To be fair, these are likely edge case scenarios.)

[1] If the user continues _without_ adding any cards, the answer to the previous required question ("Did you serve under another name?") would auto-update to be "no". I swear I saw this behavior on another single page L&L form, but of course now I can't find it.😖

**Comment by danbrady** [comment]
I see what you're saying, Todd. We might be saying the same thing. Continuing your example, how would the user remove that one/last card? In the current implementation there is not a remove button. The user would have to remember that there was a question on the _previous_ page asking if they have additional service names and update their answer there. I think we agree that that's not optimal, especially if they answered that question during a different session of filling out the form. So staging review says add a remove button even when there's one card, but it's unclear what that action should do. You think it should navigate the user back to the previous step if the user clicks "remove" on the last card and I'm thinking it should keep them on the same page to keep the "remove" actions consistent but that requires additional validation. Let me know if I'm framing that incorrectly. Allison (who submitted the issue) is out today, but I can confirm with her to see what the expected/standard behavior is from a governance standpoint.

**Comment by ToddWebDev** [comment]
I'm suggesting more discussion with staging review about the context of the multi-page form and the conditional single page L&L pattern. As a software developer, I do not want to over step boundaries so I will wait until the referenced [Figma designs](https://www.figma.com/design/9JKK5Eo43uJWEr66JPiebc/Pension-Claim-Form-21P-527EZ?node-id=5366-16247&t=BkxIMm2hsm8YyJe1-1) include a completed solution to implement.

**Comment by danbrady** [comment]
Ok, I followed up with Allison. Since this button didn't exist in single page L&L, she said we can follow what the multi-page list and loop in this [mock form for optional fields](https://staging.va.gov/mock-form-patterns/employers). So in that, if there is only one card and a user clicks to remove it, the user redirects back to the original question. I believe this is what you were thinking, Todd. I've added a note to that button in the designs to mention this behavior. Let me know if other design updates are needed. Thanks.

**Comment by ToddWebDev** [comment]
Since the multi page is now the preferred pattern for multiple responses, we will update the 'other service names' section to the new preferred pattern, rather than modifying the single page array pattern.

**Comment by ToddWebDev** [comment]
I'm moving this to the blocked lane along with the other multiple page pattern update. This will have the same associated issues as #82850 to unblock.

**Comment by gmaciolek** [comment]
Design and copy review complete. 

Found 2 issues:

1) On the summary page, "Yes" is selected by default for "Do you have another dependent child to add?" Expected behavior: neither option is selected by default
![Image](https://github.com/user-attachments/assets/4c85e916-3bb5-4da7-813f-9e6fd97bf95a)

2) In the edit previous name flow, remove the introductory sentence that reads, "We'll take you through each of the sections of this previous name for you to review and edit" (I don't think that copy is necessary here since there's only one page of information to edit.)

![Image](https://github.com/user-attachments/assets/f568fcb5-5aa6-452e-b431-d63b835c4cc3)

RE: It will be important to evaluate whether the multiple-page response is the appropriate user experience for entering multiple names, since adding a name does not require more than one page.

My thought on this: After reviewing the design system guidance and reading a few relevant Slack threads in the design system channel, I think we should move forward with this multi-page, muti-repsonse pattern here, even though we could capture the responses with the single page pattern. The multi-page solution is now the recommended approach. If we end up with research or other indicators that the pattern isn't working well here, we can consult with the design system team again.

[Figma file for reference](https://www.figma.com/design/9JKK5Eo43uJWEr66JPiebc/Pension-Claim-Form-21P-527EZ?node-id=8101-11737&t=xT5wygcXmRpgM9Os-4)

@ToddWebDev

**Comment by ToddWebDev** [comment]
I've moved this issue back to the 'Blocked' lane and added these issues to track progress on requested changes:
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/93219

**Comment by ToddWebDev** [comment]
This issue can be unblocked and is ready for content and design review.

**Comment by aplatt-coforma** [comment]
Eng work is complete and changes are on staging environment for now. 
Will pick up multiple page list & loop work once it becomes prioritized with HCD.
